pkg/storage/stores/indexshipper/downloads/table_manager.go

```diff
@@ -27,8 +25,7 @@ const (
 	daySeconds           = int64(24 * time.Hour / time.Second)
 )
 
-// regexp for finding the trailing index bucket number at the end of table name
-var extractTableNumberRegex = regexp.MustCompile(`[0-9]+$`)
+var downloadMetrics *metrics
```

Comment:
yeah, I tried using `WrapRegistererWith` but it ends up registering the metric more than once resulting in a panic.
I am looking at [CurryWith](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#HistogramVec.CurryWith), but the [NewHistogramCollector](https://github.com/grafana/loki/blob/25db6bcff287dcf6dc60a8250645573c56566b55/vendor/github.com/weaveworks/common/instrument/instrument.go#L46) used [here](https://github.com/grafana/loki/blob/25db6bcff287dcf6dc60a8250645573c56566b55/pkg/storage/stores/shipper/shipper_index_client.go#L170) doesn't accept [ObserverVec](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#ObserverVec), so I am planning to do the metric refactor in a separate pr after updating the weaveworks pkg. But even this won't get rid of the global

Reply:
I am guessing we didn't notice it so far since `index-gateway` client is suppressing it here - https://github.com/grafana/loki/blob/25db6bcff287dcf6dc60a8250645573c56566b55/pkg/storage/stores/indexshipper/gatewayclient/gateway_client.go#L120
and we've only initialized singleton botldb and tsdb index clients.

Reply:
removed the global and did a sanity check with multiple boltdb clients, it works. Not sure why it failed last time I tried